python/lsst/afw/geom/transform.cc

```diff
@@ -0,0 +1,120 @@
+*
+* This product includes software developed by the
+* LSST Project (http://www.lsst.org/).
+*
+* This program is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 3 of the License, or
+* (at your option) any later version.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the LSST License Statement and
+* the GNU General Public License along with this program. If not,
+* see <http://www.lsstcorp.org/LegalNotices/>.
+*/
+#include <memory>
+
+#include "astshim.h"
+#include "pybind11/pybind11.h"
+#include "pybind11/stl.h"
+#include "numpy/arrayobject.h"
+#include "ndarray/pybind11.h"
+
+#include "lsst/afw/geom/Endpoint.h"
+#include "lsst/afw/geom/Point.h"
+#include "lsst/afw/geom/SpherePoint.h"
+#include "lsst/afw/geom/Transform.h"
+
+namespace py = pybind11;
+using namespace py::literals;
+
+namespace lsst {
+namespace afw {
+namespace geom {
+namespace {
+
+template<typename Class>
+std::string formatStr(Class const & self, std::string const & pyClassName) {
+    std::ostringstream os;
+    os << pyClassName;
+    auto const frameSet = self.getFrameSet();
+    os << "[" << frameSet->getNin() << "->" << frameSet->getNout() << "]";
+    return os.str();
+}
+
+// Declare Transform<FromEndpoint, ToEndpoint>
+template <typename FromEndpoint, typename ToEndpoint>
+void declareTransform(py::module& mod, std::string const & fromName, std::string const & toName) {
+    using Class = Transform<FromEndpoint, ToEndpoint>;
+    using ToPoint = typename ToEndpoint::Point;
+    using ToArray = typename ToEndpoint::Array;
+    using FromPoint = typename FromEndpoint::Point;
+    using FromArray = typename FromEndpoint::Array;
+
+    std::string const pyClassName = "TransformFrom" + fromName + "To" + toName;
+
+    py::class_<Class, std::shared_ptr<Class>> cls(mod, pyClassName.c_str());
+
+    cls.def(py::init<ast::Mapping const &, bool>(), "mapping"_a, "simplify"_a=true);
+    cls.def(py::init<ast::FrameSet const &, bool>(), "frameSet"_a, "simplify"_a=true);
+
+    cls.def("getFromEndpoint", &Class::getFromEndpoint);
+    cls.def("getFrameSet", &Class::getFrameSet);
+    cls.def("getToEndpoint", &Class::getToEndpoint);
+
+    cls.def("tranForward", (ToArray (Class::*)(FromArray const &) const) &Class::tranForward, "array"_a);
+    cls.def("tranForward", (ToPoint (Class::*)(FromPoint const &) const) &Class::tranForward, "point"_a);
+    cls.def("tranInverse", (FromArray (Class::*)(ToArray const &) const) &Class::tranInverse, "array"_a);
+    cls.def("tranInverse", (FromPoint (Class::*)(ToPoint const &) const) &Class::tranInverse, "point"_a);
+    /// repr format is lsst.afw.geom.<ClassName>[<nIn>-><nOut>]
```

Comment:
Doxygen comments should not be used inside function bodies. Did you mean `//`? [Same for `__str__`]